question/bank/statistics/classes/helper.php

```diff
@@ -0,0 +1,163 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace qbank_statistics;
+
+defined('MOODLE_INTERNAL') || die();
+require_once($CFG->dirroot . '/mod/quiz/report/statistics/statisticslib.php');
+require_once($CFG->dirroot . '/mod/quiz/report/default.php');
+require_once($CFG->dirroot . '/mod/quiz/report/statistics/report.php');
+require_once($CFG->dirroot . '/mod/quiz/report/reportlib.php');
+require_once($CFG->dirroot . '/mod/quiz/attemptlib.php');
+require_once($CFG->dirroot . '/mod/quiz/classes/question/bank/qbank_helper.php');
+use core_question\statistics\questions\all_calculated_for_qubaid_condition;
+use quiz_statistics_report;
+use mod_quiz\question\bank\qbank_helper;
+/**
+ * Helper for statistics
+ *
+ * @package    qbank_statistics
+ * @copyright  2021 Catalyst IT Australia Pty Ltd
+ * @author     Nathan Nguyen <nathannguyen@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class helper {
+
+    /**
+     * Return all quizzes that use the quesion
+     *
+     * @param $questionid id of the question
+     * @return array list of quizids
+     * @throws \dml_exception
+     */
+    public static function get_quizzes($questionid): array {
+        global $DB;
+        $version = qbank_helper::get_current_version($questionid);
+
+        $quizzes = $DB->get_records_sql("
```

Comment:
what about the questions not added directly but attempted from random question loader? those need to be considered as a part part of this calculation as well